apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import sinon from 'sinon';
+import {expect} from '../../util/reconfiguredChai';
+import {mount} from 'enzyme';
+import JavalabEditor from '@cdo/apps/javalab/JavalabEditor';
+import JavalabFileManagement from '@cdo/apps/javalab/JavalabFileManagement';
+import {Provider} from 'react-redux';
+import {
+  getStore,
+  registerReducers,
+  stubRedux,
+  restoreRedux
+} from '@cdo/apps/redux';
+import javalab from '@cdo/apps/javalab/javalabRedux';
+
+describe('Java Lab Editor Test', () => {
+  let defaultProps, store;
+
+  beforeEach(() => {
+    stubRedux();
+    registerReducers({javalab});
+    store = getStore();
+    sinon.stub(JavalabFileManagement, 'renameProjectFile');
+    sinon.stub(JavalabFileManagement, 'onProjectChanged');
+    defaultProps = {
+      onCommitCode: () => {}
+    };
+  });
+
+  afterEach(() => {
+    restoreRedux();
+    JavalabFileManagement.renameProjectFile.restore();
+    JavalabFileManagement.onProjectChanged.restore();
+  });
+
+  const createWrapper = overrideProps => {
+    const combinedProps = {...defaultProps, ...overrideProps};
+    return mount(
+      <Provider store={store}>
+        <JavalabEditor {...combinedProps} />
+      </Provider>
+    );
+  };
+
+  describe('Rename', () => {
+    it('enables rename after Rename button clicked', () => {
+      const editor = createWrapper();
+
+      // find 'Rename' button and click it (enables rename)
+      const activateRenameBtn = editor.find('button').first();
+      expect(activateRenameBtn.contains('Rename')).to.be.true;
+      activateRenameBtn.invoke('onClick')();
+
+      // save button should now be second input (first is file name input)
+      const submitBtn = editor.find('input').at(1);
+      expect(submitBtn.prop('value')).to.equal('Save');
```

Comment:
the expectations of this test don't match the title of the test -- should we be checking that the button is enabled rather than (or along with) checking the value?

Reply:
good point, the test is poorly named, it's really testing if the rename button goes away and a save button apppears, hence checking the value. I'll rename the test.